File: sdkwork-backend-sdk-java/src/main/java/com/sdkwork/backend/api/ImMessageApi.java
package com.sdkwork.backend.api;

import com.sdkwork.backend.http.HttpClient;
import com.sdkwork.backend.model.*;
import java.util.List;
import java.util.Map;

public class ImMessageApi {
    private final HttpClient client;
    
    public ImMessageApi(HttpClient client) {
        this.client = client;
    }

    /** Mark conversation messages as read */
    public PlusApiResultBoolean markRead(PlusImMessageMarkReadForm body) throws Exception {
        return (PlusApiResultBoolean) client.post(ApiPaths.backendPath("/im/message/mark_read"), body);
    }

    /** Get messages by page */
    public PlusApiResultPagePlusMessage listByPage(PlusImMessageQueryForm body, Map<String, Object> params) throws Exception {
        return (PlusApiResultPagePlusMessage) client.post(ApiPaths.backendPath("/im/message/list"), body, params);
    }
}

File: sdkwork-backend-sdk-kotlin/src/main/kotlin/com/sdkwork/backend/api/ImMessageApi.kt
package com.sdkwork.backend.api

import com.sdkwork.backend.*
import com.sdkwork.backend.http.HttpClient

class ImMessageApi(private val client: HttpClient) {

    /** Mark conversation messages as read */
    suspend fun markRead(body: PlusImMessageMarkReadForm): PlusApiResultBoolean? {
        return client.post(ApiPaths.backendPath("/im/message/mark_read"), body) as? PlusApiResultBoolean
    }

    /** Get messages by page */
    suspend fun listByPage(body: PlusImMessageQueryForm? = null, params: Map<String, Any>? = null): PlusApiResultPagePlusMessage? {
        return client.post(ApiPaths.backendPath("/im/message/list"), body, params) as? PlusApiResultPagePlusMessage
    }
}
